internal/attestation/crafter/materials/csaf_security_indicent_response.go

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package materials
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	api "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/rs/zerolog"
+)
+
+type CSAFSecurityIncidentResponseCrafter struct {
+	backend *casclient.CASBackend
+	*crafterCommon
+}
+
+func NewCSAFSecurityIncidentResponseCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFSecurityIncidentResponseCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_SECURITY_INCIDENT_RESPONSE {
+		return nil, fmt.Errorf("material type is not CSAF_SECURITY_INCIDENT_RESPONSE format")
+	}
+
+	return &CSAFSecurityIncidentResponseCrafter{
+		backend:       backend,
+		crafterCommon: &crafterCommon{logger: l, input: materialSchema},
+	}, nil
+}
+
+func (i *CSAFSecurityIncidentResponseCrafter) Craft(ctx context.Context, filepath string) (*api.Attestation_Material, error) {
+	i.logger.Debug().Str("path", filepath).Msg("decoding CSAF Security Incident Response file")
+
+	fh, err := os.Open(filepath)
```

Comment:
There seems to be quite a bit of common code here. The one that opens a file from a path and marshals it into a document. 

Should we move it to the shared library `vex`?